rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+
+The importance of commit messages conveying enough context and information about
+the code being changed will grow as the project grows and bad (poorly written)
+commit messages can affect negatively the code quality of future changes that
+would otherwise benefit from past good commit messages if they existed.
+
+Beyond code itself, poorly written commit messages can also affect the community
+in other ways. For example, by not providing to new contributors a consistent
+and complete history or context for the code changes, it can work as a barrier
+for new contributions because much more time will be necessary trying to
+understand what motivated a past critical but unclear change.
+
+Hence this Commit Message Guideline can help contributors to write good commit
+messages and so improve the current situation regarding the TVM commit logs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Commit Message Guideline
+
+Apache TVM uses the Github (GH) platform for patch submission and code review
+via Pull Requests (PRs). The final commit (title and body) that is merged into
+the Apache TVM main tree is composed of the PR's title and body and must be kept
+updated and reflecting the new changes in the code as per the reviews and
+discussions.
+
+Although these guidelines apply essentially to the PRs’ title and body messages,
+because GH auto-generates the PR’s title and body from the commits on a given
+branch, it’s recommended to follow these guidelines right from the beginning,
+when preparing commits in general to be submitted to the Apache TVM project.
+This will ease the creation of a new PR, avoiding rework, and also will help the
+review.
+
+The rules below will help to achieve uniformity that has several benefits, both
+for review and for the code base maintenance as a whole, helping you to write
+commit messages with a good quality suitable for the Apache TVM project,
+allowing fast log searches, bisecting, and so on.
+
+_PR/commit title_:
+
+* Guarantee a title exists (enforced);
+* Don’t use Github usernames in the title, like @username (enforced);
+* Check if a tag should be present as a hint about what component(s) of the code
+  the commits “touch”. For example [BugFix], [CI], [microTVM], and [TVMC]. Tags
+  go between square brackets and appear first in the title. Tags help reviewers
+  to identify the PRs they can/want to review and also help the release folks
+  when generating the release notes (enforced);
+* Use an imperative mood. Avoid titles like “Added operator X” and “Updated
+  image Y in the CI”, instead use the forms “Add feature X” and “Update image Y
+  in the CI” instead;
+* Observe proper use of caps at the beginning (uppercase for the first letter)
+  and for acronyms, like, for instance, TVM, FVP, OpenCL. Hence instead of
+  “fix tvm use of opencl library”, write it as “Fix TVM use of OpenCL library”;
+* No period at the end of the title is necessary.
```

Comment:
I'm ok with the suggestion.

I just would like to clarify that this rule is not enforced, in the sense it's essentially a recommendation and won't be a blocker for the PR -- no linter will regard it, for example. In v1 we discussed that for this case it could incur an unnecessary review "overhead" by requesting to an author to change such a thing for reviewers and authors on different timezones, for instance.

Reply:
Yes, I agree with @gromero. I don't think this should be 100% prescriptive on every minimal - sometimes cosmetic - detail, so I think the original wording is OK.

Reply:
@leandron I guess what I meant is that I think @tkonolige suggestion of s/No period at the end of the title is necessary/Do not put a period at the end of the title/ is fine, but I think we should not enforce it. Are you ok if we change to the suggested wording but don't marked as "enforced"? I think the form suggested is a tad stronger but won't be 100% prescriptive if not marked as enforced.

Reply:
Yes, I don't think should be enforced.

Reply:
Addressed here: https://github.com/apache/tvm-rfcs/pull/88/commits/bb5936e231b2f0f4f156c73f5ac2fc15ba4794fe @leandron @tkonolige PTAL.

Reply:
Fine by me. I agree with the spelling suggested by @tkonolige. Marking this as resolved.